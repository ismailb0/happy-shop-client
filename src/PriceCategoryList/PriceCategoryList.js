import React from 'react';
import List from '@material-ui/core/List';
import { withStyles } from "@material-ui/core/styles";

import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Checkbox from '@material-ui/core/Checkbox';
import Collapse from '@material-ui/core/Collapse';
import ExpandLess from '@material-ui/icons/ExpandLess';
import ExpandMore from '@material-ui/icons/ExpandMore';

const RANGES = [
  {
    startPrice: 0,
    endPrice: 10,
    label: 'Under 10$',
  },
  {
    startPrice: 10,
    endPrice: 20
  },
  {
    startPrice: 20,
    endPrice: 30
  },
  {
    startPrice: 30,
    endPrice: 50
  },
  {
    startPrice: 50,
    endPrice: 100,
  },
  {
    startPrice: 100,
    endPrice: 1000000000,
    label: 'Above 100',
  },
]

const styles = {
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    paddingLeft: 5,
    paddingRight: 5,
    paddingTop: 0,
    paddingBottom: 0,
  }
};


class PriceCategoryList extends React.Component {
  state = {
    checked: [],
    opened: false
  };

  openList = () => {
    this.setState ({
      opened: !this.state.opened
    })
  }

  updateProducts = (checkedPriceRanges) => {
    let minPrice = Math.min.apply(Math, checkedPriceRanges.map((priceRange) => priceRange.startPrice));
    let maxPrice = Math.max.apply(Math, checkedPriceRanges.map((priceRange) => priceRange.endPrice));

    if(minPrice === Infinity) {
      minPrice = null
    }
    if(maxPrice === -Infinity) {
      maxPrice = null
    }

    this.props.handlePriceRangeChange(minPrice, maxPrice, checkedPriceRanges)
  }

  handleClick = value => () => {
    const { checked } = this.state;
    const currentIndex = checked.indexOf(value);
    const newChecked = [...checked];

    if (currentIndex === -1) {
      newChecked.push(value);
    } else {
      newChecked.splice(currentIndex, 1);
    }

    this.setState({
      checked: newChecked,
    });
    this.updateProducts(newChecked)
  };

  render() {
    const { classes } = this.props;

    return (
      <List component="div">
        <ListItem button onClick={() => this.openList()}>
          <ListItemText
            inset
            primary={
              <div className='menu-item-text' >Price</div>
            }
          />
          {this.state.opened ? <ExpandLess /> : <ExpandMore />}
        </ListItem>
        <Collapse
          in={this.state.opened}
          timeout="auto"
          unmountOnExit
        >
          {RANGES.map(range => (
            <ListItem
              key={range.startPrice}
              role={undefined}
              onClick={this.handleClick(range)}
            >
              <Checkbox
                className={classes.root}
                checked={this.state.checked.indexOf(range) !== -1}
                tabIndex={-1}
                disableRipple
              />
              <ListItemText
                primary={range.label
                  ? range.label
                  :`${range.startPrice}$ - ${range.endPrice}$`
              } />
            </ListItem>
          ))}
        </Collapse>
      </List>
    );
  }
}

export default withStyles(styles)(PriceCategoryList);
