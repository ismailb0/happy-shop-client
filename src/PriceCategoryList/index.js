export { default } from './PriceCategoryList.js';
