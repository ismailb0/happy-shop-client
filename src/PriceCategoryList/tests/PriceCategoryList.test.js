import React from 'react';
import { shallow } from "enzyme";
import { unwrap } from "@material-ui/core/test-utils";
import PriceCategoryList from '../PriceCategoryList';

const ComponentNaked = unwrap(PriceCategoryList);

describe('<PriceCategoryList />', () => {

  const props = {
    handlePriceRangeChange: jest.fn()
  }

  it('updateProducts should get call handlePriceRangeChange with the minPrice, maxPrice and the checked price ranges', () => {
    const renderedComponent = shallow(<ComponentNaked classes={{}} {...props} />);
    const componentInstance = renderedComponent.instance()
    const checkedPriceRanges = [
      {
        startPrice: 100,
        endPrice: 500
      },
      {
        startPrice: 680,
        endPrice: 720
      },
    ]
    componentInstance.updateProducts(checkedPriceRanges)
    expect(props.handlePriceRangeChange).toHaveBeenCalledWith(100, 720, checkedPriceRanges)
  });

  it('handleClick should get add a price range to checked if it isnt already there or remove it', () => {
    const checkedRange = {
      startPrice: 680,
      endPrice: 720
    }
    const renderedComponent = shallow(<ComponentNaked classes={{}} {...props} />);
    const componentInstance = renderedComponent.instance()
    componentInstance.state.checked = []
    jest.spyOn(componentInstance, 'updateProducts');
    componentInstance.handleClick(checkedRange)
    componentInstance.state.checked = [checkedRange]
    componentInstance.handleClick(checkedRange)
    componentInstance.state.checked = []
  });

  it('openList should get toggle the price check list', () => {
    const renderedComponent = shallow(<ComponentNaked classes={{}} {...props} />);
    const componentInstance = renderedComponent.instance()
    componentInstance.state.opened = false
    jest.spyOn(componentInstance, 'openList');
    componentInstance.openList()
    componentInstance.state.opened = true
    componentInstance.openList()
    componentInstance.state.opened = false
  });

  it('should match snapshot', () => {
    const renderedComponent = shallow(<ComponentNaked classes={{}} {...props} />);
    expect(renderedComponent).toMatchSnapshot();
  });

});
