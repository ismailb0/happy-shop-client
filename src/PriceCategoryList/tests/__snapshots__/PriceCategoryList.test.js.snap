// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriceCategoryList /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PriceCategoryList
    classes={Object {}}
    handlePriceRangeChange={
      [MockFunction] {
        "calls": Array [
          Array [
            100,
            720,
            Array [
              Object {
                "endPrice": 500,
                "startPrice": 100,
              },
              Object {
                "endPrice": 720,
                "startPrice": 680,
              },
            ],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(ListItem)
          button={true}
          onClick={[Function]}
        >
          <WithStyles(ListItemText)
            inset={true}
            primary={
              <div
                className="menu-item-text"
              >
                Price
              </div>
            }
          />
          <pure(ExpandMoreIcon) />
        </WithStyles(ListItem)>,
        <WithStyles(Collapse)
          in={false}
          timeout="auto"
          unmountOnExit={true}
        >
          <WithStyles(ListItem)
            onClick={[Function]}
          >
            <WithStyles(Checkbox)
              checked={false}
              disableRipple={true}
              tabIndex={-1}
            />
            <WithStyles(ListItemText)
              primary="Under 10$"
            />
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            onClick={[Function]}
          >
            <WithStyles(Checkbox)
              checked={false}
              disableRipple={true}
              tabIndex={-1}
            />
            <WithStyles(ListItemText)
              primary="10$ - 20$"
            />
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            onClick={[Function]}
          >
            <WithStyles(Checkbox)
              checked={false}
              disableRipple={true}
              tabIndex={-1}
            />
            <WithStyles(ListItemText)
              primary="20$ - 30$"
            />
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            onClick={[Function]}
          >
            <WithStyles(Checkbox)
              checked={false}
              disableRipple={true}
              tabIndex={-1}
            />
            <WithStyles(ListItemText)
              primary="30$ - 50$"
            />
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            onClick={[Function]}
          >
            <WithStyles(Checkbox)
              checked={false}
              disableRipple={true}
              tabIndex={-1}
            />
            <WithStyles(ListItemText)
              primary="50$ - 100$"
            />
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            onClick={[Function]}
          >
            <WithStyles(Checkbox)
              checked={false}
              disableRipple={true}
              tabIndex={-1}
            />
            <WithStyles(ListItemText)
              primary="Above 100"
            />
          </WithStyles(ListItem)>
        </WithStyles(Collapse)>,
      ],
      "component": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "button": true,
          "children": Array [
            <WithStyles(ListItemText)
              inset={true}
              primary={
                <div
                  className="menu-item-text"
                >
                  Price
                </div>
              }
            />,
            <pure(ExpandMoreIcon) />,
          ],
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "inset": true,
              "primary": <div
                className="menu-item-text"
              >
                Price
              </div>,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(ListItem)
              onClick={[Function]}
            >
              <WithStyles(Checkbox)
                checked={false}
                disableRipple={true}
                tabIndex={-1}
              />
              <WithStyles(ListItemText)
                primary="Under 10$"
              />
            </WithStyles(ListItem)>,
            <WithStyles(ListItem)
              onClick={[Function]}
            >
              <WithStyles(Checkbox)
                checked={false}
                disableRipple={true}
                tabIndex={-1}
              />
              <WithStyles(ListItemText)
                primary="10$ - 20$"
              />
            </WithStyles(ListItem)>,
            <WithStyles(ListItem)
              onClick={[Function]}
            >
              <WithStyles(Checkbox)
                checked={false}
                disableRipple={true}
                tabIndex={-1}
              />
              <WithStyles(ListItemText)
                primary="20$ - 30$"
              />
            </WithStyles(ListItem)>,
            <WithStyles(ListItem)
              onClick={[Function]}
            >
              <WithStyles(Checkbox)
                checked={false}
                disableRipple={true}
                tabIndex={-1}
              />
              <WithStyles(ListItemText)
                primary="30$ - 50$"
              />
            </WithStyles(ListItem)>,
            <WithStyles(ListItem)
              onClick={[Function]}
            >
              <WithStyles(Checkbox)
                checked={false}
                disableRipple={true}
                tabIndex={-1}
              />
              <WithStyles(ListItemText)
                primary="50$ - 100$"
              />
            </WithStyles(ListItem)>,
            <WithStyles(ListItem)
              onClick={[Function]}
            >
              <WithStyles(Checkbox)
                checked={false}
                disableRipple={true}
                tabIndex={-1}
              />
              <WithStyles(ListItemText)
                primary="Above 100"
              />
            </WithStyles(ListItem)>,
          ],
          "in": false,
          "timeout": "auto",
          "unmountOnExit": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Checkbox)
                  checked={false}
                  disableRipple={true}
                  tabIndex={-1}
                />,
                <WithStyles(ListItemText)
                  primary="Under 10$"
                />,
              ],
              "onClick": [Function],
              "role": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "checked": false,
                  "className": undefined,
                  "disableRipple": true,
                  "tabIndex": -1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "Under 10$",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "10",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Checkbox)
                  checked={false}
                  disableRipple={true}
                  tabIndex={-1}
                />,
                <WithStyles(ListItemText)
                  primary="10$ - 20$"
                />,
              ],
              "onClick": [Function],
              "role": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "checked": false,
                  "className": undefined,
                  "disableRipple": true,
                  "tabIndex": -1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "10$ - 20$",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "20",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Checkbox)
                  checked={false}
                  disableRipple={true}
                  tabIndex={-1}
                />,
                <WithStyles(ListItemText)
                  primary="20$ - 30$"
                />,
              ],
              "onClick": [Function],
              "role": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "checked": false,
                  "className": undefined,
                  "disableRipple": true,
                  "tabIndex": -1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "20$ - 30$",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "30",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Checkbox)
                  checked={false}
                  disableRipple={true}
                  tabIndex={-1}
                />,
                <WithStyles(ListItemText)
                  primary="30$ - 50$"
                />,
              ],
              "onClick": [Function],
              "role": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "checked": false,
                  "className": undefined,
                  "disableRipple": true,
                  "tabIndex": -1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "30$ - 50$",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "50",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Checkbox)
                  checked={false}
                  disableRipple={true}
                  tabIndex={-1}
                />,
                <WithStyles(ListItemText)
                  primary="50$ - 100$"
                />,
              ],
              "onClick": [Function],
              "role": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "checked": false,
                  "className": undefined,
                  "disableRipple": true,
                  "tabIndex": -1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "50$ - 100$",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "100",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Checkbox)
                  checked={false}
                  disableRipple={true}
                  tabIndex={-1}
                />,
                <WithStyles(ListItemText)
                  primary="Above 100"
                />,
              ],
              "onClick": [Function],
              "role": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "checked": false,
                  "className": undefined,
                  "disableRipple": true,
                  "tabIndex": -1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "Above 100",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(ListItem)
            button={true}
            onClick={[Function]}
          >
            <WithStyles(ListItemText)
              inset={true}
              primary={
                <div
                  className="menu-item-text"
                >
                  Price
                </div>
              }
            />
            <pure(ExpandMoreIcon) />
          </WithStyles(ListItem)>,
          <WithStyles(Collapse)
            in={false}
            timeout="auto"
            unmountOnExit={true}
          >
            <WithStyles(ListItem)
              onClick={[Function]}
            >
              <WithStyles(Checkbox)
                checked={false}
                disableRipple={true}
                tabIndex={-1}
              />
              <WithStyles(ListItemText)
                primary="Under 10$"
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              onClick={[Function]}
            >
              <WithStyles(Checkbox)
                checked={false}
                disableRipple={true}
                tabIndex={-1}
              />
              <WithStyles(ListItemText)
                primary="10$ - 20$"
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              onClick={[Function]}
            >
              <WithStyles(Checkbox)
                checked={false}
                disableRipple={true}
                tabIndex={-1}
              />
              <WithStyles(ListItemText)
                primary="20$ - 30$"
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              onClick={[Function]}
            >
              <WithStyles(Checkbox)
                checked={false}
                disableRipple={true}
                tabIndex={-1}
              />
              <WithStyles(ListItemText)
                primary="30$ - 50$"
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              onClick={[Function]}
            >
              <WithStyles(Checkbox)
                checked={false}
                disableRipple={true}
                tabIndex={-1}
              />
              <WithStyles(ListItemText)
                primary="50$ - 100$"
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              onClick={[Function]}
            >
              <WithStyles(Checkbox)
                checked={false}
                disableRipple={true}
                tabIndex={-1}
              />
              <WithStyles(ListItemText)
                primary="Above 100"
              />
            </WithStyles(ListItem)>
          </WithStyles(Collapse)>,
        ],
        "component": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "button": true,
            "children": Array [
              <WithStyles(ListItemText)
                inset={true}
                primary={
                  <div
                    className="menu-item-text"
                  >
                    Price
                  </div>
                }
              />,
              <pure(ExpandMoreIcon) />,
            ],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "inset": true,
                "primary": <div
                  className="menu-item-text"
                >
                  Price
                </div>,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(ListItem)
                onClick={[Function]}
              >
                <WithStyles(Checkbox)
                  checked={false}
                  disableRipple={true}
                  tabIndex={-1}
                />
                <WithStyles(ListItemText)
                  primary="Under 10$"
                />
              </WithStyles(ListItem)>,
              <WithStyles(ListItem)
                onClick={[Function]}
              >
                <WithStyles(Checkbox)
                  checked={false}
                  disableRipple={true}
                  tabIndex={-1}
                />
                <WithStyles(ListItemText)
                  primary="10$ - 20$"
                />
              </WithStyles(ListItem)>,
              <WithStyles(ListItem)
                onClick={[Function]}
              >
                <WithStyles(Checkbox)
                  checked={false}
                  disableRipple={true}
                  tabIndex={-1}
                />
                <WithStyles(ListItemText)
                  primary="20$ - 30$"
                />
              </WithStyles(ListItem)>,
              <WithStyles(ListItem)
                onClick={[Function]}
              >
                <WithStyles(Checkbox)
                  checked={false}
                  disableRipple={true}
                  tabIndex={-1}
                />
                <WithStyles(ListItemText)
                  primary="30$ - 50$"
                />
              </WithStyles(ListItem)>,
              <WithStyles(ListItem)
                onClick={[Function]}
              >
                <WithStyles(Checkbox)
                  checked={false}
                  disableRipple={true}
                  tabIndex={-1}
                />
                <WithStyles(ListItemText)
                  primary="50$ - 100$"
                />
              </WithStyles(ListItem)>,
              <WithStyles(ListItem)
                onClick={[Function]}
              >
                <WithStyles(Checkbox)
                  checked={false}
                  disableRipple={true}
                  tabIndex={-1}
                />
                <WithStyles(ListItemText)
                  primary="Above 100"
                />
              </WithStyles(ListItem)>,
            ],
            "in": false,
            "timeout": "auto",
            "unmountOnExit": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Checkbox)
                    checked={false}
                    disableRipple={true}
                    tabIndex={-1}
                  />,
                  <WithStyles(ListItemText)
                    primary="Under 10$"
                  />,
                ],
                "onClick": [Function],
                "role": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "checked": false,
                    "className": undefined,
                    "disableRipple": true,
                    "tabIndex": -1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "primary": "Under 10$",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "10",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Checkbox)
                    checked={false}
                    disableRipple={true}
                    tabIndex={-1}
                  />,
                  <WithStyles(ListItemText)
                    primary="10$ - 20$"
                  />,
                ],
                "onClick": [Function],
                "role": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "checked": false,
                    "className": undefined,
                    "disableRipple": true,
                    "tabIndex": -1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "primary": "10$ - 20$",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "20",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Checkbox)
                    checked={false}
                    disableRipple={true}
                    tabIndex={-1}
                  />,
                  <WithStyles(ListItemText)
                    primary="20$ - 30$"
                  />,
                ],
                "onClick": [Function],
                "role": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "checked": false,
                    "className": undefined,
                    "disableRipple": true,
                    "tabIndex": -1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "primary": "20$ - 30$",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "30",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Checkbox)
                    checked={false}
                    disableRipple={true}
                    tabIndex={-1}
                  />,
                  <WithStyles(ListItemText)
                    primary="30$ - 50$"
                  />,
                ],
                "onClick": [Function],
                "role": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "checked": false,
                    "className": undefined,
                    "disableRipple": true,
                    "tabIndex": -1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "primary": "30$ - 50$",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "50",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Checkbox)
                    checked={false}
                    disableRipple={true}
                    tabIndex={-1}
                  />,
                  <WithStyles(ListItemText)
                    primary="50$ - 100$"
                  />,
                ],
                "onClick": [Function],
                "role": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "checked": false,
                    "className": undefined,
                    "disableRipple": true,
                    "tabIndex": -1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "primary": "50$ - 100$",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "100",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Checkbox)
                    checked={false}
                    disableRipple={true}
                    tabIndex={-1}
                  />,
                  <WithStyles(ListItemText)
                    primary="Above 100"
                  />,
                ],
                "onClick": [Function],
                "role": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "checked": false,
                    "className": undefined,
                    "disableRipple": true,
                    "tabIndex": -1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "primary": "Above 100",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
